### ISSUE TITLE
s25rttr-gui is shattered with modern opengl merged on linux

### ISSUE THREAD

**Issue by stefson** [Return-To-The-Roots/s25client#890]
Not sure how to give this a catchy headline, but: with the modern opengl branch merged into the s25client repo, I can't run the gui of /usr/bin/s25rttr anymore. The gui flickers, is black, and jumps very fast from one corner of the screen to another.

/usr/bin/s25edit is working (in the sense that it displays a gui)

a quick bisect shows that most likely 341a43f3fef0682ab9f290fbe6cd01472d20848b introduced this: 

```

Return To The Roots v20181101-1868b25e6d2fd0a359af5764b67baaf2d8002f6a
Compiled with Clang (6.0.1) for Linux, Unix Environment 64 Bit

Starting in /usr
Suche nach Treibern in /usr/lib64/s25rttr/video
1 video drivers found!
Loaded video driver "(SDL) OpenGL via SDL-Library"
OpenGL 4.4 supported
Suche nach Treibern in /usr/lib64/s25rttr/audio
1 audio drivers found!
Loaded audio driver "(SDL) Audio via SDL_mixer-Library"

Starte Spiel
Lade "/usr/share/s25rttr/RTTR/splash.bmp": fertig in 10ms
Lade "/usr/share/s25rttr/S2/GFX/PALETTE/PAL5.BBM": fertig in 0ms
Lade "/usr/share/s25rttr/S2/GFX/PALETTE/PAL6.BBM": fertig in 0ms
Lade "/usr/share/s25rttr/S2/GFX/PALETTE/PAL7.BBM": fertig in 0ms
Lade "/usr/share/s25rttr/S2/GFX/PALETTE/PALETTI0.BBM": fertig in 0ms
Lade "/usr/share/s25rttr/S2/GFX/PALETTE/PALETTI1.BBM": fertig in 0ms
Lade "/usr/share/s25rttr/S2/GFX/PALETTE/PALETTI8.BBM": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/COLORS.ACT": fertig in 0ms
Lade "/usr/share/s25rttr/S2/DATA/RESOURCE.DAT": fertig in 7ms
Lade "/usr/share/s25rttr/S2/DATA/IO/IO.DAT": fertig in 3ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP013.LBM": fertig in 8ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP015.LBM": fertig in 8ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP666.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP667.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP801.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP802.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP803.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP804.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP805.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP806.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP810.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP811.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP895.LBM": fertig in 8ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/SETUP896.LBM": fertig in 8ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/AFRICA.LBM": fertig in 7ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/AUSTRA.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/EUROPE.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/GREEN.LBM": fertig in 8ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/JAPAN.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/NAMERICA.LBM": fertig in 8ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/NASIA.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/SAMERICA.LBM": fertig in 9ms
Lade "/usr/share/s25rttr/S2/GFX/PICS/MISSION/SASIA.LBM": fertig in 9ms
Lade "/home/tim/.s25rttr/LSTS/SOUND.LST": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0001.OGG": fertig in 1ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0002.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0003.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0004.OGG": fertig in 1ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0005.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0006.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0007.OGG": fertig in 1ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0008.OGG": fertig in 1ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0009.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0010.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0011.OGG": fertig in 1ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0012.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0013.OGG": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/MUSIC/SNG/SNG_0014.OGG": fertig in 1ms
Loading LST,LBM,BOB,IDX,BMP,TXT,GER,ENG,INI files from "/usr/share/s25rttr/RTTR/LSTS"
Lade "/usr/share/s25rttr/RTTR/LSTS/CLIENT.GER": fertig in 0ms
Lade Ordner /usr/share/s25rttr/RTTR/LSTS/CREDITS.LST
  Lade "0" Einträge: fertig in 2ms
Lade "/usr/share/s25rttr/RTTR/LSTS/OUTLINE_FONTS.LST": fertig in 2ms
Lade "/usr/share/s25rttr/RTTR/LSTS/RTTR.LST": fertig in 0ms
Lade Ordner /usr/share/s25rttr/RTTR/LSTS/firework.LST
  Lade "32" Einträge: fertig in 18ms
Lade Ordner /usr/share/s25rttr/RTTR/LSTS/io_new.LST
  Lade "14" Einträge: fertig in 5ms
Lade "/usr/share/s25rttr/RTTR/LSTS/languages.ini": fertig in 0ms
Lade "/usr/share/s25rttr/RTTR/LSTS/logo.bmp": fertig in 1ms
Lade "/usr/share/s25rttr/RTTR/LSTS/menu.bmp": fertig in 42ms
fertig in 70ms
Loading LST,LBM,BOB,IDX,BMP,TXT,GER,ENG,INI files from "/home/tim/.s25rttr/LSTS"
fertig in 0ms
Loading "/usr/share/s25rttr/RTTR/MUSIC/S2_Standard.pll"
^CDo you really want to terminate the program (y/n) : y
^CDo you really want to kill the program (y/n) : y
Benchmark #1 took 0 seconds -> 0 milliseconds/frame
Benchmark #2 took 0 seconds -> 0 milliseconds/frame
Benchmark #3 took 0 seconds -> 0 milliseconds/frame
Benchmark #4 took 0 seconds -> 0 milliseconds/frame
Benchmark #5 took 0 seconds -> 0 milliseconds/frame
Total benchmark time; 0 seconds -> 0 milliseconds/frame
Saeubere Texturespeicher: fertig (nach 0ms)

```

I can't stop it by `Strg + C` , have to kill /usr/bin/s25rttr via taskmanager. The Benchmark message won't be displayed before s25client got hit by `SIGTERM`

**Comment by Flamefire** [comment]
First, seems to be the buggy benchmark (strange that I did not see this on windows) can you further check if it is indeed this commit?

2nd: Should be fixed along with #881

**Comment by stefson** [comment]
This was likely a bug in mesa-18.2.2, update came in today for 18.2.4 and with the fix from #881 I got it all back to work. 

Do you still allow building against system lua with dyn. linked .so files?
